packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/utils/synthesize-resources.test.ts

```diff
@@ -23,11 +23,28 @@ describe('correctly updates userPool group list', () => {
   });
   afterEach(() => jest.resetAllMocks());
 
+  const expectAmplifyMetaFileUpdate = (): void => {
+    expect(mockContext.amplify.updateamplifyMetaAfterResourceUpdate).toBeCalledTimes(3);
```

Comment:
other option is e2e test...

Reply:
Did you find any tests that 'read' the file which is written to by 'synthesize-resources.ts'? I am curious if we are missing a test to verify reads/writes to the file... or how the error slipped in

Reply:
I have not unfortunately. I took a quick stab at what it would look like to use real updateamplifyMetaAfterResourceUpdate but that means dragging cli-internal on dependency path :-(